src/modules/get_professors_by_class/app/get_professors_by_class_usecase.ts

```diff
@@ -0,0 +1,15 @@
+
+import { IScheduleRepository } from '../../../shared/domain/repositories/schedule_repository_interface';
+import { NoItemsFound } from '../../../shared/helpers/errors/usecase_errors'
+
+export class GetProfessorsByClassUsecase {
+  constructor(private repo: IScheduleRepository) {}
+
+  async execute(classId: string) {
+    const professors = await this.repo.getProfessorsByClass(classId);
+    if (!professors.length) {
```

Comment:
return empty list when class doesn't have professors (its not an error on Database the existance of a class without professors in certain moment)

Reply:
This conflicts when the class doesn't have any professors, and when the classId is not found